src/sagemaker/base_deserializers.py

```diff
@@ -0,0 +1,324 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""Implements base methods for deserializing data returned from an inference endpoint."""
+from __future__ import absolute_import
+
+import csv
+
+import abc
+import codecs
+import io
+import json
+
+import numpy as np
+from six import with_metaclass
+
+from sagemaker.utils import DeferredError
+
+try:
+    import pandas
+except ImportError as e:
+    pandas = DeferredError(e)
+
+
+class BaseDeserializer(abc.ABC):
+    """Abstract base class for creation of new deserializers.
+
+    Provides a skeleton for customization requiring the overriding of the method
+    deserialize and the class attribute ACCEPT.
+    """
+
+    @abc.abstractmethod
+    def deserialize(self, stream, content_type):
+        """Deserialize data received from an inference endpoint.
+
+        Args:
+            stream (botocore.response.StreamingBody): Data to be deserialized.
+            content_type (str): The MIME type of the data.
+
+        Returns:
+            object: The data deserialized into an object.
+        """
+
+    @property
+    @abc.abstractmethod
+    def ACCEPT(self):
+        """The content types that are expected from the inference endpoint."""
+
+
+class SimpleBaseDeserializer(with_metaclass(abc.ABCMeta, BaseDeserializer)):
+    """Abstract base class for creation of new deserializers.
+
+    This class extends the API of :class:~`sagemaker.deserializers.BaseDeserializer` with more
+    user-friendly options for setting the ACCEPT content type header, in situations where it can be
+    provided at init and freely updated.
```

Comment:
`SimpleBaseDeserializer` does not really communicate intent, and the less inheritance depth we have the better. We may want to consider if this class is really necessary as I think it is fine for each concrete subclass of `BaseDeserializer` to set `accept`.